amazonka/src/Network/AWS/Auth.hs

```diff
@@ -175,11 +180,12 @@ data Credentials
     | Discover
       -- ^ Attempt credentials discovery via the following steps:
       --
-      -- * Read the 'envAccessKey' and 'envSecretKey' from the environment if they are set.
+      -- * Read the 'envAccessKey', 'envSecretKey', and 'envRegion' from the environment if they are set.
```

Comment:
It's something I'm working towards - it is a CLI thing but the other SDKs all support it via the `AWS_SDK_LOAD_CONFIG` flag.